### PR TITLE
Authorisation architecture ADR

### PR DESCRIPTION
**Pull request by zackads**
See the doc! 😁

### CODE REVIEW

File: docs/adr/0006-2021-05-26-authorisation.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    beacon records.
+3.  Application code in the Web App will restrict the permissions of the current Beacon Owner / public account
+    holder to operate only on beacon records within their scope.
+4.  Maritime & Coastguard Agency internal users will authenticate using Azure AD to access the Beacons [Back Office
+    Admin Tool](https://github.com/mcagov/beacons-backoffice). The Maritime & Coastguard Agency's production tenant
+    will be used to allow Single Sign-On. Using Azure AD will allow role-based access control of back office
+    operations, giving the ability to make Search & Rescue users 'read only' while allowing Beacon Registration Team
+    users to add flags, comments and other metadata.
+
+![Beacons authorisation diagram](assets/beacons-auth-diagram.png).
```
Comment (matthew-a-carr) [inline docs/adr/0006-2021-05-26-authorisation.md:39]: Is this a png extension?  Not sure if that will open up the excalidraw picture

File: docs/adr/0006-2021-05-26-authorisation.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  the Web App must be implemented in the application.
+  - The confidential client key held by the Web App represents a larger blast radius in the event of a compromised
+    credential than the alternative delegation model (see alternative).
+
+## Alternatives
+
+_Summarise the main alternatives that have been considered and the reasons why they have not been chosen. Use no
+more than one paragraph per alternative._
+
+1.  **Full OAuth 2.0 delegation model**. The beacon owner / account holder is made a true OAuth 2.0 resource owner that
```
Comment (matthew-a-carr) [inline docs/adr/0006-2021-05-26-authorisation.md:67]: I would add to this that this current option isn't possible due to B2C not supporting on behalf of user flows

**Review by matthew-a-carr — APPROVED** [review]
@zackads LGTM, just a few changes to make

File: docs/adr/0006-2021-05-26-authorisation.md (hunk 1 above)
Comment (zackads) [reply docs/adr/0006-2021-05-26-authorisation.md:39]: This is correct.  There's a .png in there for display in the ADR and the .excalidraw file is included if we want to use it for a future ADR / update it.

Comment (matthew-a-carr) [reply docs/adr/0006-2021-05-26-authorisation.md:39]: Ah yes, sorry missed that

File: docs/adr/0006-2021-05-26-authorisation.md (hunk 2 above)
Comment (zackads) [reply docs/adr/0006-2021-05-26-authorisation.md:67]: Ah I think this option is possible with B2C? 

The [on-behalf-of](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow) user flow would be required if there was a protected resource at an API behind our API, so the chain would go `resource owner (user) > client (webapp) > intermediary (registrations API) -> protected resource (another API)`.

image

What I mean by the full OAuth 2.0 delegation model is the standard authorization code grant type (`resource owner (user) -> client (webapp) -> protected resource (registrations API)`).  

I _hope_ I've got this right 😅. I've added a couple of sentences to the option to clarify.

**Pull request merged**